### DISCUSSION TITLE
Markdownの画像記法に属性を含められるようにしたい

### DISCUSSION THREAD

**Discussion by yarakos95** [increments/qiita-discussions#452] [要望・アイデア・不具合]
現在、Qiita に画像をアップロードすると、次のような Markdown 形式で文章内に挿入されます。

```markdown
![filename.png](https://~~~.png)
``` 

しかし、この画像のサイズを変更するために `width` 等の属性を加えようと思うと、`<img>` タグに修正する必要があります。これは非常に面倒であり、出来る限り Markdown で済ませたいと思ってしまいます。

このことから、Pandoc's Markdown に代表されるような、`{}` による属性を追加できるオプション記法があると良いと思います。

https://pandoc.org/MANUAL.html#extension-link_attributes

Pandoc ではいくつかの属性種を含めることが出来ますが、Qiita では `width`・`height` 等の大きさを変更するための属性のみが含められれば十分だと思います。したがって、希望したい記法は次のようになります。

```markdown
![filename.png](https://~~~.png) { width=80% }

![filename.png](https://~~~.png "image description") { width=80% height=80% }

![filename.png][ref]

[ref]: https://~~~.png "image description" { width=80% }
```

**Comment by tanaka-asami415** [comment]
@Yarakashi-Kikohshi 
ありがとうございます🙏
要望は多くいただいており、我々としてもニーズがあることは理解しているのですが、、、

現在、QiitaのMarkdown記法はGitHub Flavored Markdownがベースになっています。
様々なMarkdownを使用してしまうと、Qiitaのエディタ以外で執筆する方の互換性がなくなってしまうこともあり
現時点で実装をお約束することは難しい、、、という回答になります。申し訳ございません😢

**Reply by yarakos95** [reply]
@tanaka-asami415 さま

> 現在、QiitaのMarkdown記法はGitHub Flavored Markdownがベースになっています。
> 様々なMarkdownを使用してしまうと、Qiitaのエディタ以外で執筆する方の互換性がなくなってしまうこともあり

GFM との互換性の考慮から、複数の Markdown 記法を利用することを避けたいことは理解できます。しかし、以下 2 点の理由から、互換性の考慮によってリジェクトされるには至らないのではないかと思います。

### 1. 大きく互換性を損なわないのでは？

提案した記法では、GFM による画像挿入の互換性を著しく損なうものではないと思っています。`{}` によるオプション記法は、これを理解しないパーサにとってはリテラルに解釈されるものになります。私も Qiita エディタ以外で編集する一人ですが、このような画像の後ろにリテラルに解釈される部分があっても、それほど気にならないと思います。

```markdown
<!-- Markdown -->
![filename.png](link/image.png) { width=80% }
``` 
　↓
```html
<!-- HTML -->
<img src="link/image.png" alt="filename.png"> { width=80% }
``` 

大きさを変更しない場合には `{}` を付けなければ良いだけなので、互換性に関する問題が増えるものではないと考えています。

比較検討として、一部では `![filename.png](link/image.png 80x50)` のような記法も存在しています。この記法では、GFM による画像挿入との互換性を著しく損なってしまいます。今回の提案記法では、それほどの問題を生じないと思っています。

### 2. 互換性は自力で確保すれば良いのでは？

Markdown の互換性に関する私個人の意見として、互換性は自力で確保していくものだと考えています。

実際、Qiita では GFM にない Note 記法があります。細かいところを突けば、シンタックスハイライトで指定する言語名は GitHub と異なるものがいくつかあります。shortcodes を使った絵文字 (`:shortcodes:`) も、完全に一致するものではありません。（`:person_frowning:` は、Qiita では利用できますが、GitHub では利用できない）

これらのことを踏まえると、互換性は自力で確保していけば良く、提案した記法が許されていても、各個人が`<img>` タグを利用して画像の大きさを調整すれば問題がないと思います。

Qiita ではすでに GitHub Markdown との互換性は取れていると考えています。これを侵さないレベルの記法の追加は、利用者の裁量によって制限すれば良いだろうと思います。

**Reply by tanaka-asami415** [reply]
@Yarakashi-Kikohshi 
ありがとうございます！🙏
ご提案いただいた方法で大丈夫そうではあるのですが、
記法の拡張は一度やると元に戻すことが難しいため、テストしながら慎重に行いたいと思っています。

直近の開発優先度などから考えると先になってしまうのと
ご提案いただいた通りの実装とはならない場合もあるかもしれませんが、その点はご容赦いただけますと幸いです🙏

**Reply by yarakos95** [reply]
@tanaka-asami415 さま

> 直近の開発優先度などから考えると先になってしまうのと
ご提案いただいた通りの実装とはならない場合もあるかもしれませんが、その点はご容赦いただけますと幸いです🙏

 提案したような記法が新たに追加される日を楽しみにしています！！
